src/services/users/users.services.js

```diff
@@ -0,0 +1,46 @@
+// const userData = {
+//   name: 'Homer Simpson',
+// };
+
+// let users = Array.from({ length: 7 }, (_, id) => ({ id, ...userData }));
+let users = [];
+
+let lastId = users.length;
+
+const getUsers = () => users;
+
+const saveUser = (name) => {
+  lastId += 1;
+
+  const newUser = { id: lastId, name };
+
+  users.push(newUser);
+
+  return newUser;
+};
+
+const getUserById = (id) => getUsers().find((user) => user.id === id) || null;
+
+const updateUser = (id, name) => {
+  const user = getUserById(id);
+
+  if (!user) {
+    throw new Error('No User');
+  }
+
+  Object.assign(user, { name });
+
+  return { id, name };
+};
+
+const removeUser = (userId) => {
+  users = getUsers().filter(({ id }) => id !== userId);
```

Comment:
The `removeUser` function is correctly filtering the `users` array, but ensure that this operation is atomic if used in a concurrent environment to avoid potential race conditions.